nni/retiarii/execution/base.py

```diff
@@ -93,18 +95,21 @@ def _final_metric_callback(self, trial_id: int, metrics: MetricData) -> None:
         for listener in self._listeners:
             listener.on_metric(model, metrics)
 
-    def query_available_resource(self) -> List[WorkerInfo]:
-        raise NotImplementedError  # move the method from listener to here?
+    def query_available_resource(self) -> int:
+        return self.resources
 
     @classmethod
     def trial_execute_graph(cls) -> None:
         """
         Initialize the model, hand it over to trainer.
         """
         graph_data = BaseGraphData.load(receive_trial_parameters())
-        with open('_generated_model.py', 'w') as f:
+        random_str = ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(6))
+        file_name = f'_generated_model_{random_str}.py'
```

Comment:
Might be better if we put all temp files into a folder called `generated` or `.retiarii` or something...

Reply:
good point, will update

Reply:
in the next pr